azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java

```diff
@@ -87,10 +88,13 @@ public void run() throws Exception {
           break;
         }
         if (attempt < ServerInternals.MEMORY_CHECK_RETRY_LIMIT) {
```

Comment:
One thing still confusing me is that There looks duplicate checking logics in Line90 and line 84.
`attempt <= ServerInternals.MEMORY_CHECK_RETRY_LIMIT`

Are they really necessary?